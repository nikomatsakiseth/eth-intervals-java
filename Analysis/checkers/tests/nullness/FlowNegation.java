public class FlowNegation {

    void testSimpleValid() {
        String s = "m";
        s.toString();
    }

    void testCase1() {
        String s = "m";
        if (s != null) { }
        else { }
        s.toString();
    }

    void testCase2() {
        String s = "m";
        if (s == null) { }
        else { }
        s.toString();
    }

    void testInvalidCase1() {
        String s = "m";
        if (s != null) { s = null; }
        else { }
        s.toString();   // error
    }

    void testInvalidCase2() {
        String s = "m";
        if (s != null) { }
        else { s = null; }
        s.toString();   // error
    }

    void testSimpleValidTernary() {
        String s = "m";
        s.toString();
    }

    void testTernaryCase1() {
        String s = "m";
        Object m = (s != null) ? "m" : "n";
        s.toString();
    }

    void testTernaryCase2() {
        String s = "m";
        Object m = (s == null) ? "m" : "n";
        s.toString();
    }

    void testTernaryInvalidCase1() {
        String s = "m";
        Object m = (s != null) ? (s = null) : "n";
        s.toString();   // error
    }

    void testTernaryInvalidCase2() {
        String s = "m";
        Object m = (s != null) ? "m" : (s = null);
        s.toString();   // error
    }

    void testAssignInCond() {
        String s = "m";
        if ((s = null) != "m") {
            s.toString();   // error
        } else {
        }
        s.toString();   // error
    }
}
