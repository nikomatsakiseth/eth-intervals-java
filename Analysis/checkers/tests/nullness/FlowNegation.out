:26: (dereference.of.nullable)
:33: (dereference.of.nullable)
:56: (dereference.of.nullable)
:62: (dereference.of.nullable)
:68: (dereference.of.nullable)
:71: (dereference.of.nullable)
